### PR TITLE
Fix mistake (?) in IEnumerable<T> documentation.

### PR DESCRIPTION
**Pull request by m-ringler**
AFAIS, a reference to the non-generic interface was intended by the original author.

### CODE REVIEW

**Comment by m-ringler** [comment]
The section is hard to understand anyways. 'parity' is not commonly used for what is probably meant here.

I'd probably prefer 'compatibility with non-generic APIs' to 'parity with non-generic collections'.

**Comment by mairaw** [comment]
@safern can you review this PR?

**Comment by safern** [comment]
This change makes sense to me. Thanks @m-ringler. 

> compatibility with non-generic APIs

I also agree this would make it simpler to understand, @mairaw what do you think?

File: xml/System.Collections.Generic/IEnumerable`1.xml [hunk 1]
```diff
@@ -64,7 +64,7 @@
     </remarks>
     <block subset="none" type="overrides">
       <para>
-        <see cref="T:System.Collections.Generic.IEnumerable`1" /> is included for parity with non-generic collections; implementing <see cref="T:System.Collections.Generic.IEnumerable`1" /> allows a generic collection to be passed to a method that expects an <see cref="T:System.Collections.IEnumerable" /> object.</para>
+        <see cref="T:System.Collections.IEnumerable" /> is included for parity with non-generic collections; implementing <see cref="T:System.Collections.Generic.IEnumerable`1" /> allows a generic collection to be passed to a method that expects an <see cref="T:System.Collections.IEnumerable" /> object.</para>
```
Comment (rpetrusha) [inline xml/System.Collections.Generic/IEnumerable`1.xml]:
```suggestion
        To remain compatible with methods that iterate non-generic collections, <see cref="T:System.Collections.Generic.IEnumerable`1" /> implements <see cref="T:System.Collections.IEnumerable" />. This allows a generic collection to be passed to a method that expects an <see cref="T:System.Collections.IEnumerable" /> object.</para>
```

**Comment by m-ringler** [comment]
Thanks @rpetrusha , that makes it a lot easier to understand.

**Comment by safern** [comment]
Thanks @m-ringler for noticing and updating.

File: xml/System.Collections.Generic/IEnumerable`1.xml [hunk 2]
```diff
@@ -64,7 +64,7 @@
     </remarks>
     <block subset="none" type="overrides">
       <para>
```
Comment (mairaw) [inline xml/System.Collections.Generic/IEnumerable`1.xml]:
```suggestion
```

File: xml/System.Collections.Generic/IEnumerable`1.xml [hunk 3]
```diff
@@ -64,7 +64,7 @@
...
-        <see cref="T:System.Collections.Generic.IEnumerable`1" /> is included for parity with non-generic collections; implementing <see cref="T:System.Collections.Generic.IEnumerable`1" /> allows a generic collection to be passed to a method that expects an <see cref="T:System.Collections.IEnumerable" /> object.</para>
+        To remain compatible with methods that iterate non-generic collections, <see cref="T:System.Collections.Generic.IEnumerable`1" /> implements <see cref="T:System.Collections.IEnumerable" />. This allows a generic collection to be passed to a method that expects an <see cref="T:System.Collections.IEnumerable" /> object.</para>
```
Comment (mairaw) [inline xml/System.Collections.Generic/IEnumerable`1.xml]: We could combine the para tag line with the text in one line if we're not starting with a tag.

```suggestion
      <para>To remain compatible with methods that iterate non-generic collections, <see cref="T:System.Collections.Generic.IEnumerable`1" /> implements <see cref="T:System.Collections.IEnumerable" />. This allows a generic collection to be passed to a method that expects an <see cref="T:System.Collections.IEnumerable" /> object.</para>
```

File: xml/System.Collections.Generic/IEnumerable`1.xml (hunk 2 above)
Comment (m-ringler) [reply xml/System.Collections.Generic/IEnumerable`1.xml]: Fine for me.

**Comment by rpetrusha** [comment]
Thanks for the additional changes, @m-ringler. We'll merge when the build completes successfully.

**Pull request merged**